persistence/core/src/multi-jvm/scala/com/lightbend/lagom/internal/persistence/cluster/ClusterDistributionSpec.scala

```diff
@@ -0,0 +1,185 @@
+      case -1 => s
+      case z  => s.drop(z + 1)
+    }
+    reduced.head.replaceFirst(""".*\.""", "").replaceAll("[^a-zA-Z_0-9]", "_")
+  }
+  def createActorSystem(): Config => ActorSystem = { config =>
+    val setup = ActorSystemSetup(BootstrapSetup(ConfigFactory.load(config)))
+    ActorSystem(getCallerName(classOf[MultiNodeSpec]), setup)
+  }
+}
+
+class ClusterDistributionSpecMultiJvmNode1 extends ClusterDistributionSpec
+class ClusterDistributionSpecMultiJvmNode2 extends ClusterDistributionSpec
+class ClusterDistributionSpecMultiJvmNode3 extends ClusterDistributionSpec
+
+class ClusterDistributionSpec
+    extends MultiNodeSpec(ClusterDistributionConfig, ClusterDistributionSpec.createActorSystem())
+    with STMultiNodeSpec
+    with ImplicitSender {
+
+  import ClusterDistributionConfig._
+
+  //  -------- A first piece of testing scaffolding and cluster formation  ------------------
+  override def initialParticipants: Int = roles.size
+
+  def join(from: RoleName, to: RoleName): Unit = {
+    runOn(from) {
+      Cluster(system).join(node(to).address)
+    }
+    enterBarrier(from.name + "-joined")
+  }
+
+  def fullAddress(ref: ActorRef): Address =
+    if (ref.path.address.hasLocalScope) Cluster(system).selfAddress
+    else ref.path.address
+
+  protected override def atStartup() {
+    roles.foreach(n => join(n, node1))
+    within(15.seconds) {
+      awaitAssert(Cluster(system).state.members.size should be(3))
+      awaitAssert(
+        Cluster(system).state.members.toIndexedSeq.map(_.status).distinct should be(IndexedSeq(MemberStatus.Up))
+      )
+    }
+
+    enterBarrier("startup")
+  }
+
+  //  -------- Before this line all code is testing scaffolding or cluster formation  ------------------
+
+  private val ensureActiveInterval: FiniteDuration = 1.second
+  val distributionSettings =
+    ClusterDistributionSettings(system)
+      .copy(ensureActiveInterval = ensureActiveInterval)
+
+  "A ClusterDistribution" must {
+
+    "distribute the entityIds across nodes (so all nodes get a response)" in {
+
+      val probe = TestProbe()
+
+      val typeName = "CDTest"
+      // There'll be 3 nodes in this test and each node will host a TestProbe.
+      // Cluster Distribution on each node will create a `FakeActor.props` pointing
+      // back to its own TestProbe. We request the creation of 10 FakeActor to ClusterDistribution
+      // with the expectation that there'll be 3 or 4 FakeActor's for each TestProbe.
+      val props: Props = FakeActor.props(probe.ref)
+      val entityIds    = (1 to 10).map(i => s"test-entityId$i").toSet
+
+      // Load the extension and wait for other nodes to be ready before proceeding
+      val cdExtension  = ClusterDistribution(system)
+      enterBarrier("extension-is-loaded")
+      cdExtension.start(
+        typeName,
+        props,
+        entityIds,
+        distributionSettings
+      )
+
+      // During 10secs, or when 10 responses are obtained, record responses.
```

Comment:
The comment says 10 responses. The code on line 154 says 20.

Reply:
Good catch. I had to increase the number for travis. 

In Travis, everything happens more slowly so remote nodes are penalized. As a consequence, waiting for only 10 messages didn't leave enough time for all nodes to report back.

Reply:
```suggestion
      // During 10secs, or when 20 responses are obtained, record responses.
```